f5_openstack_agent/lbaasv2/drivers/bigip/network_service.py

```diff
@@ -411,13 +410,16 @@ def create_rd_by_net(
 
                 LOG.info("create route domain: %s, %s on bigip: %s"
                          % (name, route_domain, bigip.hostname))
-        except Exception as ex:
-            if ex.response.status_code == 409:
+        except HTTPError as err:
+            if err.response.status_code == 409:
                 LOG.info("route domain %s already exists: %s, ignored.." % (
-                    route_domain, ex.message))
+                    route_domain, err.message))
+            elif err.response.status_code == 400:
```

Comment:
Does this change improve anything?

Reply:
if it is 400, we throw up the originall HTTPError, rather than f5_ex.RouteDomainCreationException.